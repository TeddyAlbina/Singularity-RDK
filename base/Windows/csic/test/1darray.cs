public class test {
	bool[] x = new bool[10];
	public static void Main() {
		test t = new test();
		Emit(t.x[1]);
	}
	public static void Emit(bool flag) {
		System.Console.WriteLine(flag);
	}
}
