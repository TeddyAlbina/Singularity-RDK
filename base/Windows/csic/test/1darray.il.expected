.assembly '1darray' { }
.assembly extern 'mscorlib' {
.ver 1:0:5000:0
.publickeytoken = ( b7 7a 5c 56 19 34 e0 89 )
}
.class public 'test'
	extends [mscorlib]System.Object
{
.field private bool[] 'x'
.method hidebysig public static void 'Main'() {
.entrypoint
.locals init ([0]class 'test' 't0')
newobj instance void 'test'::.ctor()
stloc 0 // t
ldloc 0 // t
ldfld bool[] 'test'::'x'
ldc.i4 1
ldelem.i1
call  void 'test'::'Emit'(bool)
ret
.maxstack 2
} // end of method test.Main
.method hidebysig public static void 'Emit'(bool 'flag') {
ldarg 0 // flag
call  void [mscorlib]System.Console::'WriteLine'(bool)
ret
.maxstack 1
} // end of method test.Emit
.method hidebysig specialname rtspecialname public void .ctor() {
ldarg.0
ldc.i4 10
newarr bool
stfld bool[] 'test'::'x'
ldarg.0
call instance void [mscorlib]System.Object::.ctor()
ret
.maxstack 2
} // end of constructor test.test
} // end of class test
